### PR TITLE
Merge MicroPython v1.11 into CircuitPython

### PR DESCRIPTION
**Pull request by tannewt**
Another step for #2999

### CODE REVIEW

File: py/circuitpy_defns.mk [hunk 1]
```diff
@@ -639,6 +639,19 @@ $(addprefix lib/,\
...
+	lib/oofatfs/ff.c \
+	lib/oofatfs/ffunicode.c \
+	lib/timeutils/timeutils.c \
+	lib/utils/buffer_helper.c \
+	lib/utils/context_manager_helpers.c \
+	lib/utils/interrupt_char.c \
+	lib/utils/pyexec.c \
+	lib/utils/stdout_helpers.c \
+	lib/utils/sys_stdio_mphal.c
+
```
Comment (dhalbert) [inline py/circuitpy_defns.mk:654]: Yay!

**Review by jepler — DISMISSED** [review]
I looked at a subset of this and didn't see anything of concern.  No testing performed.

**Review by dhalbert — COMMENTED** [review]
I looked through all the interesting files (I skip the tests). Looks good, and this is simpler than V1.11, so it should get easier and easier.

I'll do some more extensive filesystem testing when the build works, due to the update of `oofatfs`.

**Comment by tannewt** [comment]
Ok, build is passing. I turned off some modules to make it fit. Mostly synthio, msgpack, and asyncio.

**Review by dhalbert — APPROVED** [review]
Works for me! I used the Metro M4 version, edited code.py, and also copied and editing a much larger file, whose contents were fine over reboots.

**Pull request merged**